src/utils/assert_op.jl

```diff
@@ -0,0 +1,31 @@
+"""
+Throw an `AssertionError` if `op(exp1, exp2)` is `false`.
+
+Use only the forms below. `@assert_op a == b` is not currently supported.
```

Comment:
I was not able to easily figure how to support this. If you prefer that this be addressed before we add it to the codebase, I'm OK with it.

Reply:
Use this implementation:

```
function assert_op(expr::Expr)
    @assert expr.head == :call
    @assert length(expr.args) == 3

    return :(
        if $(expr.args[1])($(esc(expr.args[2])), $(esc(expr.args[3])))
            return nothing
        else
            name1 = $(string(expr.args[2]))
            name2 = $(string(expr.args[3]))
            val1 = $(esc(expr.args[2]))
            val2 = $(esc(expr.args[3]))
            throw(AssertionError("$name1 = $val1, $name2 = $val2"))
        end
    )

end

macro assert_op(expr)
    return assert_op(expr)
end
```

See output below:

```
$ julia --startup-file=no --project
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.5.2 (2020-09-23)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> include("test.jl")
@assert_op (macro with 1 method)

julia> a = 3; b = 4;

julia> @assert_op a == b
ERROR: AssertionError: a = 3, b = 4
Stacktrace:
 [1] top-level scope at /Users/USER/gitrepos/pyhelics/test.jl:13

julia> @assert_op (a + 1) == b

julia> @assert_op a + 1 == b

julia> @assert_op a + 3 == b + 4
ERROR: AssertionError: a + 3 = 6, b + 4 = 8
Stacktrace:
 [1] top-level scope at /Users/USER/gitrepos/pyhelics/test.jl:13
```

Reply:
This will end up with output like `4 = 4` though. We'll have to write more special casing logic to handle that.

```
julia> @assert_op a == 3

julia> @assert_op a == 4
ERROR: AssertionError: a = 3, 4 = 4
Stacktrace:
 [1] top-level scope at /Users/USER/gitrepos/repo/test.jl:13
```

Reply:
What do you think of this output instead?
```
julia> @assert_op a == 4
ERROR: AssertionError: 3 is not == 4
Stacktrace:
 [1] top-level scope at /home/dthom/sandboxes/InfrastructureSystems.jl/src/utils/assert_op.jl:49
```

Reply:
@kdheepak Thanks for the enhancement. Note that both of our implementations had a bug, though. It was introducing a return into the code. I fixed that here. 

I'll wait for your comment about the new error message.

Reply:
Resolved in https://github.com/NREL-SIIP/InfrastructureSystems.jl/pull/172/commits/1635b68985e61fbdd100daabecc9263c7deea53a.